Sources/ArcGISToolkit/FloorFilter/FloorFilterViewController.swift

```diff
@@ -0,0 +1,536 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+import Foundation
+
+
+
+protocol FloorFilterViewControllerDelegate {
+    // Updates the Floor Filter levels list with the site and facility that were selected from the prompt
+    func siteFacilityIsUpdated(viewModel: FloorFilterViewModel)
+}
+
+
+public class FloorFilterViewController: UIViewController, FloorFilterViewControllerDelegate {
+    /// The direction the floor filter should expand in.
+    public enum Style {
+        /// The level list expands down from the floor filter button.
+        case expandDown
+        /// The level list expands up from the floor filter button.
+        case expandUp
+    }
+    
+    /// The style of the floor filter.  The default is `.expandUp`.
+    public var style: Style = .expandUp
+    
+    /// Public variables and functions accessible to the developer
+    private var _selectedSite: AGSFloorSite? = nil
+    public var selectedSite: AGSFloorSite? {
+        get {
+            _selectedSite ?? viewModel.selectedSite
+        }
+        set {
+            _selectedSite = newValue
+            selectedFacility = nil
+            selectedLevel = nil
+            viewModel.selectedSite = newValue
+            viewModel.zoomToSelection()
+        }
+    }
+    
+    private var _selectedFacility: AGSFloorFacility? = nil
+    public var selectedFacility: AGSFloorFacility? {
+        get {
+            _selectedFacility ?? viewModel.selectedFacility
+        }
+        set {
+            _selectedFacility = newValue
+            if (_selectedFacility != nil) {
+                _selectedSite = viewModel.selectedFacility?.site
+            }
+            selectedLevel = viewModel.getDefaultLevelForFacility(facility: newValue)
+            viewModel.selectedFacility = newValue
+            viewModel.zoomToSelection()
+        }
+    }
+    
+ 
+    private var _selectedLevel: AGSFloorLevel? = nil
+    public var selectedLevel: AGSFloorLevel? {
+        get {
+            _selectedLevel ?? viewModel.selectedLevel
+        }
+        set {
+            if (_selectedLevel != newValue) {
+                _selectedLevel = newValue
+            }
+            if (_selectedLevel != nil) {
+                let selectedLevelsFacility = viewModel.selectedLevel?.facility
+                _selectedSite = selectedLevelsFacility?.site
+            }
+            viewModel.selectedLevel = newValue
+            viewModel.filterMapToSelectedLevel()
+        }
+    }
```

Comment:
These can be used in the example if the developer wants to set a selectedLevel. Removing this, the developer won't have access to directly change the selected site/facility/level.

Reply:
If you want the user to be able to modify them, I would consider making the `FloorFilterViewModel` public and moving this functionality into that class, which already has these three properties.

Reply:
@mhdostal making the `FloorFilterViewModel` public could be an option, however, that would expose some of the private functions and logic to the user that includes zooming in, access to the data that is used in the UI, etc. Additionally, it might add some complication for the user to have to set up the `FloorFilterViewModel `and pass it as an argument. On our initial design, we had submitted these variables to be public. And we wanted to keep the `FloorFilterViewController` initialization with just a geoView. @tomhillgeo please feel free to add any of your thoughts as well.

Reply:
Ok.